src/extension.ts

```diff
@@ -102,6 +102,11 @@ export function activate(context: vscode.ExtensionContext) {
                 vscode.window.showInformationMessage('Stack Protocol Server is starting, please try again later!');
             }
         });
+        
+        vscode.commands.registerCommand('adapter.restartserver',  async () => {
```

Comment:
add `context' parameter which is going to be server selected in Servers view. If 'context' is not null that means action was called from context menu for specific item in Servers view. Use it to get serverId and serverType from `serverData.servers`.

if `context` is not defined, then use code to show quick pick as you already did.